monitor/services/src/main/java/org/datacleaner/monitor/configuration/ComponentsCache.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.monitor.configuration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Class ComponentsCache
+ * For caching and storing Components and configuration
+ *
+ * @author k.houzvicka
+ * @since 24.7.15
+ */
+public class ComponentsCache {
+    private static final Logger logger = LoggerFactory.getLogger(ComponentsCache.class);
+
+    ConcurrentHashMap<String, ComponentsCacheConfigWrapper> data = new ConcurrentHashMap<>();
```

Comment:
After getting familiar with Guava Cache options and after a team discussion we decided not to use Guava Cache and keep the original implementation of the cache system.

1) Guava Cache does not provide a way how to set various timeout values for various elements in the store. AFAIK there is only one global setting for the whole cache object.
2) Our cache handles two sets of data. The first one is the configuration set in the memory which could have been covered by Guava Cache. The second is the repository set which is stored on the disk. We need the possibility to handle both sets independently. For example remove only one of them for the specified key (instanceId).